app/utils/EventReportingHelper.scala

```diff
@@ -21,26 +21,21 @@ import identifiers.EventReportingId
 import models.{EventReporting, ListOfSchemes}
 import models.requests.AuthenticatedRequest
 import play.api.libs.json.{JsValue, Json}
-import play.api.mvc.Call
 import uk.gov.hmrc.http.{HeaderCarrier, SessionKeys}
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object EventReportingHelper {
 
   final case class EventReportingData(data:EventReporting, sessionStorageKey:String)
-  def eventReportingData(srn:String, listOfSchemes: ListOfSchemes, schemeName:String, returnUrl: Call)
+
+  def eventReportingData(srn:String, listOfSchemes: ListOfSchemes, dataNoPstr: String => EventReporting)
```

Comment:
Could this be renamed? It appears to be a function that maps a pstr to an object so the name doesn't make sense to me

Reply:
Any suggestions on how this could be renamed? 
It is essentially EventReporting instance with pstr missing.
pstr => EventReporting(pstr, a, b, c) etc

Reply:
Perhaps something like pstrToEventReporting